custom_components/waste_collection_schedule/waste_collection_schedule/source/ryde_nsw_gov_au.py

```diff
@@ -0,0 +1,118 @@
+import datetime
+import json
+
+import requests
+from bs4 import BeautifulSoup
+from requests.utils import requote_uri
+from waste_collection_schedule import Collection
+
+TITLE = "City of Ryde (NSW)"
+DESCRIPTION = "Source for City of Ryde rubbish collection."
+URL = "https://www.ryde.nsw.gov.au/"
+TEST_CASES = {
+    "Ryde Aquatic Centre": {
+        "post_code": "2112",
+        "suburb": "Ryde",
+        "street_name": "Victoria Road",
+        "street_number": "504",
+    },
+    "Harris Farm Markets Boronia Park": {
+        "post_code": "2111",
+        "suburb": "Gladesville",
+        "street_name": "Pittwater Road",
+        "street_number": "128",
+    },
+    "Eastwood Shopping Centre": {
+        "post_code": "2122",
+        "suburb": "Eastwood",
+        "street_name": "Rowe Street",
+        "street_number": "152",
+    },    
+}
+
+API_URLS = {
+    "address_search": "https://www.ryde.nsw.gov.au/api/v1/myarea/search?keywords={}",
+    "collection": "https://www.ryde.nsw.gov.au/ocapi/Public/myarea/wasteservices?geolocationid={}&ocsvclang=en-AU"
+}
+
+HEADERS = {"user-agent": "Mozilla/5.0"}
+
+ICON_MAP = {
+    "General Waste": "trash-can",
+    "Recycling": "mdi:recycle",
+    "Green Waste": "mdi:leaf",
+}
+
+class Source:
+    def __init__(
+        self, post_code: str, suburb: str, street_name: str, street_number: str
+    ):
+        self.post_code = post_code
+        self.suburb = suburb
+        self.street_name = street_name
+        self.street_number = street_number
+
+    def fetch(self):
+        locationId = 0
+
+        address = "{} {} {} {}".format(
+            self.street_number, self.street_name, self.suburb, self.post_code
+        )
+
+        q = requote_uri(str(API_URLS["address_search"]).format(address))
+
+        # Retrieve suburbs
+        r = requests.get(q, headers=HEADERS)
+
+        data = json.loads(r.text)
+
+        # Find the ID for our suburb
+        for item in data["Items"]:
+            locationId = item["Id"]
+            break
+
+        if locationId == 0:
+            return []
```

Comment:
Please raise an Exception here as people using the integration should know what went wrong (Exceptions will show in the logs) instead of just getting an empty calendar

Reply:
Thanks for the review - I've updated both of these.